testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -105,6 +112,67 @@ def test_check_set_unit_nolength(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+# 82 -> 84%; lines 757-758
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        assert D._check_set_unit(units) == None
+
+    # line 774
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        D._check_set_unit(units)
+        del D.units['density']
+        D._check_set_unit({})
+        assert D.units['density'] == None
+
+    # 84 -> 86%; lines 789-792
+    def test_parameters_isdensity(self, D):
+        with pytest.warns(UserWarning, match='Running make_density()'):
+            D.make_density()
+
+    # 86 -> 89%; lines 869-871
+    def test_check_convert_density_grid_not_density(self, D1):
+        unit = 'nm^{-3}'
+        with pytest.raises(RuntimeError, match="The grid is not a density"):
+            D1.convert_density(unit)
+
+    # 89 -> 92%; lines 874-880
+    def test_check_convert_density_value_error(self, D):
+        unit = 'A^{-2}'
+        with pytest.raises(ValueError, match="The name of the unit"):
+            D.convert_density(unit)
+
+    # 92 -> 93%; line 872
+    def test_check_convert_density_units_same_density_units(self, D):
+        unit = 'A^{-3}'
+        D.convert_density(unit)
+        assert unit == D.units['density']
+
+    # 93 -> 93%; line 881
+    def test_check_convert_density_units_density(self, D):
+        unit = 'nm^{-3}'
+        D.convert_density(unit)
+        assert D.units['density'] == unit
+
+    # 93 -> 95%; line 822
+    def test_convert_length_same_length_units(self, D):
+        unit = 'A'
+        D.convert_length(unit)
+        assert unit == D.units['length']
+
+    # 95 -> 97%; line 884
+    def test_parameters_is_Density(self, D, D1):
+        D.__repr__()
+        D1.__repr__()
+        assert 'isDensity' in D1.parameters
+
+    # 97 -> 100%; lines 824-827
+    def test_check_convert_length(sef, D):
+        unit='nm'
+        D.convert_length(unit)
+        D._update()
```

Comment:
Doesn't `convert_length` already call `_update` if it's not the same units?

Reply:
Yes, my bad, now it's gone.